### DISCUSSION TITLE
Generate Metadata Not Sufficient.

### DISCUSSION THREAD

**Discussion by Lucien950** [vercel/next.js#51007] [Ideas]
### Goals

1. Provide an alternative where data fetched in the main server component can be passed to the metadata

### Background

Consider a server component with some form of query in the url, whether param query or path query. Based on the URL query, I want to display a different set of data. I want this to also be reflected in the title of the page, which is controlled by the metadata object (or generateMetadata function). However, at the moment I would need refetch the data in the generateMetadata function to display the title. I would like to not have to do this.

### Proposal

I am not sure how metadata is implemented, but if you could return the metadata in the page component that would satisfy the requirements. Or if you could control on of the parameters for the generateMetadat a function.

**Comment by tine-krueger** [comment]
If you call the fetch-function internally, the request is cached so nothing is refetched. You will find the explanation [here](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#caching-data)
